src/Provider/CommonVCS/VCSEngine.ts

```diff
@@ -1,88 +1,66 @@
 import { VCS } from '../@interfaces/VCS';
-import { LogSeverity, LogType } from '../../Parser';
+import { LogType } from '../../Parser';
 import { Log } from '../../Logger';
-import { MessageUtil } from '../utils/message.util';
-import { onlyIn, onlySeverity } from '../utils/filter.util';
-import { groupComments } from '../utils/commentUtil';
-import { Comment } from '../@types/CommentTypes';
-import { Diff } from '../@types/PatchTypes';
+import { Comment } from '../../AnalyzerBot/@types/CommentTypes';
+import { VCSEngineConfig } from '../@interfaces/VCSEngineConfig';
+import { AnalyzerBot } from '../../AnalyzerBot/AnalyzerBot';
+import { VCSAdapter } from '../@interfaces/VCSAdapter';
 
-export abstract class VCSEngine implements VCS {
-  private touchedDiff: Diff[];
-  private comments: Comment[];
-  protected nWarning: number;
-  protected nError: number;
-  protected constructor(
-    private readonly removeOldComment: boolean = false,
-    private readonly failOnWarnings: boolean = false,
+export class VCSEngine implements VCS {
+  protected analyzerBot: AnalyzerBot;
+  constructor(
+    private readonly config: VCSEngineConfig,
+    private readonly adapter: VCSAdapter,
   ) {}
 
-  abstract vcsGetLatestCommitSha(): string;
-  abstract vcsDiff(): Promise<Diff[]>;
-  abstract vcsCreateComment(comment: string): Promise<void>;
-  abstract vcsCreateReviewComment(
-    text: string,
-    file: string,
-    line: number,
-  ): Promise<void>;
-
-  abstract vcsRemoveExistingComments(): Promise<void>;
-  abstract vcsName(): string;
-
   async report(logs: LogType[]): Promise<boolean> {
     try {
+      await this.adapter.init();
       await this.setup(logs);
 
-      if (this.removeOldComment) {
-        await this.vcsRemoveExistingComments();
+      if (this.config.removeOldComment) {
+        await this.adapter.removeExistingComments();
       }
 
-      await Promise.all(this.comments.map((c) => this.createReviewComment(c)));
+      await Promise.all(
+        this.analyzerBot.comments.map((c) => this.createReviewComment(c)),
+      );
       await this.createSummaryComment();
 
       Log.info('Report commit status completed');
     } catch (err) {
-      Log.error(`${this.vcsName()} report failed`, err);
+      Log.error(`${this.adapter.getName()} report failed`, err);
       throw err;
     }
-    return this.failOnWarnings ? this.nError + this.nWarning === 0 : this.nError === 0;
+    return await this.adapter.wrapUp(this.analyzerBot);
+  }
+
+  private async setup(logs: LogType[]) {
+    const touchedDiff = await this.adapter.diff();
+    this.analyzerBot = new AnalyzerBot(this.config, logs, touchedDiff);
```

Comment:
not sure we should make classes instantiate any object by themselves. it would make VCSEngine harder to test as it couple.